orchagent/main.cpp

```diff
@@ -227,61 +206,73 @@ int main(int argc, char **argv)
         }
     }
 
-    SWSS_LOG_NOTICE("Notify syncd INIT_VIEW");
-
     attr.id = SAI_REDIS_SWITCH_ATTR_NOTIFY_SYNCD;
     attr.value.s32 = SAI_REDIS_NOTIFY_SYNCD_INIT_VIEW;
     status = sai_switch_api->set_switch_attribute(gSwitchId, &attr);
 
     if (status != SAI_STATUS_SUCCESS)
     {
-        SWSS_LOG_ERROR("Failed to notify syncd INIT_VIEW %d", status);
+        SWSS_LOG_ERROR("Failed to notify syncd INIT_VIEW, rv:%d", status);
         exit(EXIT_FAILURE);
     }
-
-    SWSS_LOG_NOTICE("Enable redis pipeline");
+    SWSS_LOG_NOTICE("Notify syncd INIT_VIEW");
 
     attr.id = SAI_REDIS_SWITCH_ATTR_USE_PIPELINE;
     attr.value.booldata = true;
 
     status = sai_switch_api->set_switch_attribute(gSwitchId, &attr);
     if (status != SAI_STATUS_SUCCESS)
     {
-        SWSS_LOG_ERROR("Failed to enable redis pipeline %d", status);
+        SWSS_LOG_ERROR("Failed to enable redis pipeline, rv:%d", status);
         exit(EXIT_FAILURE);
     }
+    SWSS_LOG_NOTICE("Enable redis pipeline");
 
-    status = sai_switch_api->create_switch(&gSwitchId, switch_attrs.size(), switch_attrs.data());
-    if (status != SAI_STATUS_SUCCESS)
-    {
-        SWSS_LOG_ERROR("Failed to create a switch %d", status);
-        exit(EXIT_FAILURE);
-    }
+    vector<sai_attribute_t> attrs;
+
+    attr.id = SAI_SWITCH_ATTR_INIT_SWITCH;
+    attr.value.booldata = true;
+    attrs.push_back(attr);
+
+    attr.id = SAI_SWITCH_ATTR_FDB_EVENT_NOTIFY;
+    attr.value.ptr = (void *)on_fdb_event;
+    attrs.push_back(attr);
+
+    attr.id = SAI_SWITCH_ATTR_PORT_STATE_CHANGE_NOTIFY;
+    attr.value.ptr = (void *)on_port_state_change;
+    attrs.push_back(attr);
+
+    attr.id = SAI_SWITCH_ATTR_SHUTDOWN_REQUEST_NOTIFY;
+    attr.value.ptr = (void *)on_switch_shutdown_request;
+    attrs.push_back(attr);
 
     attr.id = SAI_SWITCH_ATTR_SRC_MAC_ADDRESS;
-    if (!gMacAddress)
+    if (gMacAddress)
+    {
+        memcpy(attr.value.mac, gMacAddress.getMac(), 6);
+        attrs.push_back(attr);
+    }
+    else
     {
         status = sai_switch_api->get_switch_attribute(gSwitchId, 1, &attr);
         if (status != SAI_STATUS_SUCCESS)
         {
-            SWSS_LOG_ERROR("Failed to get MAC address from switch %d", status);
+            SWSS_LOG_ERROR("Failed to get MAC address from switch, rv:%d", status);
             exit(EXIT_FAILURE);
         }
         else
         {
             gMacAddress = attr.value.mac;
         }
```

Comment:
fixed.

---
In reply to: [127554898](https://github.com/Azure/sonic-swss/pull/253#discussion_r127554898) [](ancestors = 127554898)